cmd/cert.go

```diff
@@ -0,0 +1,122 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rancher/rke/cluster"
+	"github.com/rancher/rke/hosts"
+	"github.com/rancher/rke/k8s"
+	"github.com/rancher/rke/log"
+	"github.com/rancher/rke/pki"
+	"github.com/rancher/rke/services"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/urfave/cli"
+	"k8s.io/client-go/util/cert"
+)
+
+func CertCommand() cli.Command {
+	certFlags := []cli.Flag{
+		cli.StringFlag{
+			Name:   "config",
+			Usage:  "Specify an alternate cluster YAML file",
+			Value:  pki.ClusterConfig,
+			EnvVar: "RKE_CONFIG",
+		},
+	}
+
+	certFlags = append(certFlags, commonFlags...)
+
+	return cli.Command{
+		Name:   "cert-rotate",
+		Usage:  "rotate all certificates without ca",
+		Action: clusterRotateFromCli,
+		Flags:  certFlags,
+	}
+}
+
+func ClusterRotate(
+	ctx context.Context,
+	rkeConfig *v3.RancherKubernetesEngineConfig,
+	dockerDialerFactory, localConnDialerFactory hosts.DialerFactory,
+	k8sWrapTransport k8s.WrapTransport, configDir string) (string, string, string, string, map[string]pki.CertificatePKI, error) {
+
+	log.Infof(ctx, "Rotating Kubernetes cluster certificates")
+	var APIURL, caCrt, clientCert, clientKey string
+	kubeCluster, err := cluster.ParseCluster(ctx, rkeConfig, clusterFilePath, configDir, dockerDialerFactory, localConnDialerFactory, k8sWrapTransport)
+	if err != nil {
+		return APIURL, caCrt, clientCert, clientKey, nil, err
+	}
+
+	err = kubeCluster.TunnelHosts(ctx, false)
+	if err != nil {
+		return APIURL, caCrt, clientCert, clientKey, nil, err
+	}
+
+	currentCluster, err := kubeCluster.GetClusterState(ctx)
+	if err != nil {
+		return APIURL, caCrt, clientCert, clientKey, nil, err
+	}
+
+	if currentCluster == nil {
+		log.Infof(ctx, "This is a newely built cluster, can't rotate certificates")
+		return APIURL, caCrt, clientCert, clientKey, nil, nil
+	}
+
+	err = cluster.RotateRKECertificates(ctx, kubeCluster, currentCluster)
+	if err != nil {
+		return APIURL, caCrt, clientCert, clientKey, nil, err
+	}
+
+	err = kubeCluster.SetUpHosts(ctx, true)
```

Comment:
Nit: you might use `if err := <>; err != nil` construct on this line, and lines 51, 66, 85